src/components/main-view/main-view.jsx

```diff
@@ -16,107 +20,138 @@ export class MainView extends React.Component {
-  }
-  goBack() {
-    this.setState({ selectedMovie: null });
-  }
-  onLoggedIn(user) {
-    this.setState({ user });
-  }
-  startRegister() {
-    this.setState({ register : true });
-  }
-  endRegister() {
-    this.setState({ register : null });
+  
+  // props.onLoggedIn(data) - LoginView
+  onLoggedIn(authData) {
+    console.log(authData);
+    this.setState({
+      user: authData.user.Username,
+    });
+
+    localStorage.setItem("token", authData.token);
+    localStorage.setItem("user", authData.user.Username);
+    this.getMovies(authData.token);
   }
 
-  render() {
-    const { movies, selectedMovie, user, register } = this.state;
+  onLoggedOut() {
+		localStorage.removeItem('token');
+		localStorage.removeItem('user');
+		this.setState({
+			user: null,
+		});
+	}
 
-   if (register) {
-      return <RegistrationView endRegister={() => this.endRegister()} />;
-    } 
-
-    if (!user)
-      return  <LoginView startRegister={() => this.startRegister()}
-      onLoggedIn={(user) => this.onLoggedIn(user)}/>
+  render() {
+    const { movies, user } = this.state;
 
     // Before the movies have been loaded
     if (!movies) return <div className="main-view" />;
 
     return (
-      <div className="main-view">
-      <Container>
-      <Row className="justify-content-md-center">
-        {selectedMovie ? (          
-            <Col md={8}>
-          <MovieView movie={selectedMovie} goBack={() => this.goBack()} />
-          </Col>
-        
-        ) : (
-          movies.map((movie) => (
-            <MovieCard
-              key={movie._id}
-              movie={movie}
-              onClick={(movie) => this.onMovieClick(movie)}
-            />
-          ))
-        )}
-        </Row>
-      </Container>
+      <Router>
+        <div className="main-view">
+          <h2>Main View</h2>
+
+        <ul>
+          <li>
+          <Route path="/register" render={() => <RegistrationView />} />
+          <Link to="/register">Register</Link>
+          </li>
+
+        <li>
+          <Route exact path="/movies/:movieId" render={({match}) => 
+          <MovieView movie={movies.find(m => m._id === match.params.movieId)}/>}/>
+          <Link to="/movies/:movieId">Movies</Link>
+        </li>
+         
+         
+        <li>
+          <Route exact path="/"
+            render={() => 
+            {if (!user) return (
+            <LoginView onLoggedIn={(user) => this.onLoggedIn(user)}/> );
+            return movies.map((m) => 
+            <MovieCard key={m._id} movie={m} />);
+            }}/>
+
+           <Link to="/">All List</Link>
+        </li>
+
+        <li>
+            <Route exact path="/genres/:name" render={({match}) => {
+             if (!movies) return <div className="main-view" />;
+             return <GenreView genre={movies.find((m) => 
+            m.Genre.Name === match.params.name).Genre}/>;
+          }}/>
+         <Link to="/genres/:name">Genres</Link>
```

Comment:
This does not work this way. React needs to know here what the parameter `:name` is (what genre do you want to render data from).